### ISSUE TITLE
Display the status of a layer using symbol generator in layer list

### ISSUE THREAD

**Issue by rbonnefoi** [qgis/QGIS#51458]
### Feature description

Hi all,

I recently started using symbol generator. In the layer list, the corresponding layer has no symbol (only the check box to display or undisplay), while more classical layers show the symbol legend.

I undestand it is impossible to display a kind of legend for layers using generated symbol, but maybe a special icon could inform that it is using indeed symbol generation.

Moreover, when the layer is in edit mode, the small yellow pen isn't shown aside the layer name, so it is hard to rapidly check whether the edit mode is active and/or whether last changes have been saved.

In conclusion, in my opinion, it would be useful to :
1. show the editing status of a layer using the symbol generator in the layer list,
2. show a specific icon (pseudo-legend) saying the layer uses symbol generator (and maybe the kind of source geometry and target geometry, if readable).

Regards.
